pkg/pfcpsim/pfcpsim.go

```diff
@@ -121,6 +121,52 @@ func (c *PFCPClient) receiveFromN4() {
 	}
 }
 
+// ListenN4 allows for UP initiated PFCP associations. It waits for a connection and for an association request from UP.
+func (c *PFCPClient) ListenN4() error {
+	conn, err := net.ListenUDP("udp", &net.UDPAddr{IP: net.ParseIP("0.0.0.0"), Port: PFCPStandardPort})
+	if err != nil {
+		return err
+	}
+	// Expect a request in 10 seconds, otherwise close connection.
+	if err = conn.SetDeadline(time.Now().Add(10 * time.Second)); err != nil {
+		return err
+	}
+
+	buf := make([]byte, 1500)
+	n, _, err := conn.ReadFrom(buf)
+
+	if err != nil {
+		if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
+			return NewTimeoutExpiredError(netErr)
+		}
+
+		return err
+	}
+
+	msg, err := message.Parse(buf[:n])
+	if err != nil {
+		return err
+	}
+
+	if _, ok := msg.(*message.AssociationSetupRequest); !ok {
```

Comment:
we're not saving UP's node ID anywhere, for now. I assume that we should do that in future